File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int letterCount(string text)
{
    // Calculate the length of the text
    int length = strlen(text);

    // Count the alphabetic characters
    int letters = 0;

    for (int i = 0; i < length; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }

    return letters;
}

int wordCount(string text)
{
    // Count the spaces to determine the number of words
    int words = 0;
    int length = strlen(text);

    for (int i = 0; i < length; i++)
    {
        if (isspace(text[i]))
        {
            words++;
        }
    }

    return words + 1;
}

int sentCount(string text)
{
    // Count the sentence-ending punctuation marks
    int length = strlen(text);
    int sent = 0;

    for (int i = 0; i < length; i++)
    {
        if (text[i] == '!' || text[i] == '.' || text[i] == '?')
        {
            sent++;
        }
    }

    return sent;
}

int main(void)
{
    // Prompt the user for input
    string text = get_string("Text: ");

    // Get the counts
    int letterCnt = letterCount(text);
    int wordCnt = wordCount(text);
    int sentCnt = sentCount(text);

    // Calculate average sentence length and average letters per word
    double avgS = 100.0 * ((double) sentCnt / wordCnt);
    double avgL = 100.0 * ((double) letterCnt / wordCnt);

    // Calculate the Coleman-Liau index
    double index = 0.0588 * avgL - 0.296 * avgS - 15.8;

    // Print the reading level
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %.0f\n", roundf(index));
    }
}

File: week6/sentimental-readability/readability.py
# TODO
from cs50 import get_string

# converted my c code into python


def letter_count(text):
    # Calculate the length of the text
    length = len(text)

    # Count the alphabetic characters
    letters = 0

    for char in text:
        if char.isalpha():
            letters += 1

    return letters


def word_count(text):
    # Count the spaces to determine the number of words
    words = 0
    for char in text:
        if char.isspace():
            words += 1

    return words + 1


def sentence_count(text):
    # Count the sentence-ending punctuation marks
    sent = 0
    for char in text:
        if char in ['!', '.', '?']:
            sent += 1

    return sent


def main():
    # Prompt the user for input
    text = get_string("Text: ")

    # Get the counts
    letter_cnt = letter_count(text)
    word_cnt = word_count(text)
    sent_cnt = sentence_count(text)

    # Calculate average sentence length and average letters per word
    avg_s = 100.0 * (sent_cnt / word_cnt)
    avg_l = 100.0 * (letter_cnt / word_cnt)

    # Calculate the Coleman-Liau index
    index = 0.0588 * avg_l - 0.296 * avg_s - 15.8

    # Print the reading level
    if index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {round(index)}")


if __name__ == "__main__":
    main()
